File: 0189-rotate-array/0189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        k %= n
        if k == 0:
            return

        def reverse(left, right):
            while left < right:
                nums[left], nums[right] = nums[right], nums[left]
                left += 1
                right -= 1
       
        reverse(0, n - 1)
        reverse(0, k - 1)
        reverse(k, n - 1)

File: 0189-rotate-array/0189-rotate-array.swift
class Solution {
    func rotate(_ nums: inout [Int], _ k: Int) {
        let n = nums.count

        if k == 0 || k == n {
            return
        }

        let k = k % n

        nums.reverse()
        nums[0..<k].reverse()
        nums[k..<n].reverse()
        
    }
}